flake.nix

```diff
@@ -0,0 +1,59 @@
+{
+  inputs = {
+    nixpkgs.url = "github:nixos/nixpkgs/nixos-23.11";
+
+    fenix = {
+      url = "github:nix-community/fenix";
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
+
+    flake-compat = {
+      url = "github:edolstra/flake-compat";
+      flake = false;
+    };
+  };
+
+  outputs = { self, nixpkgs, fenix, ... }:
+    let
+      inherit (nixpkgs) lib;
+
+      supportedSystems = [
+        "x86_64-linux"
+        "aarch64-linux"
+        "x86_64-darwin"
+        "aarch64-darwin"
+      ];
+
+      eachSupportedSystem = lib.genAttrs supportedSystems;
+
+      mkDevShells = system:
+        let
+          pkgs = import nixpkgs { inherit system; };
+
+          rust-toolchain = fenix.packages.${system}.fromToolchainFile {
+            file = ./rust-toolchain.toml;
+            sha256 = "opUgs6ckUQCyDxcB9Wy51pqhd0MPGHUVbwRKKPGiwZU=";
+          };
+
+          defaultShell = pkgs.mkShell {
+            buildInputs = with pkgs; [
+              # Nix
+              nixd
+              nixpkgs-fmt
+
+              # Rust
+              rust-toolchain
+              cargo-watch
```

Comment:
Can we be a bit more mindful here? Is it easy to add more tools to a dev shell? If so, we probably should add only tools here that are required to fully check/build the project, i.e., the things that just will install?

Reply:
Same as `nixd`, it is for convenience. It can be removed. Later on we can add another dev shell with more complete dependencies.